### DISCUSSION TITLE
guide-R/donnees_ponderees/regression-logistique-binaire-ponderee

### DISCUSSION THREAD

**Discussion by giscus** [larmarange/guide-R#47] [Forum]
# guide-R/donnees_ponderees/regression-logistique-binaire-ponderee

https://larmarange.github.io/guide-R/donnees_ponderees/regression-logistique-binaire-ponderee.html

**Comment by RaphaelFen** [comment]
Bonjour,
J'ai une question concernant l'affichage des résultats. Dans le cadre d'une régression logistique multivariée (quasibinomial) sur données pondérées, est-il possible d'afficher dans un tbl_regression() les crude OR + IC 95% aux côtés des adjusted OR + IC95% ?  (Un peu ce que l'on retrouve dans la sortie d'un logistic.display())
Un immense merci pour votre travail et votre partage

**Reply by larmarange** [reply]
Il faut utiliser `tbl_uvregression()` pour obtenir les coefficients bruts.

Pour un exemple de tableau avec l'ensemble des résultats, voir https://larmarange.github.io/guide-R/analyses/regression-logistique-binaire.html#pr%C3%A9senter-lensemble-des-r%C3%A9sultats-dans-un-m%C3%AAme-tableau

C'est un exemple sur des données non pondérées. Il faut l'adapter au cas pondéré mais la logique est la même.

**Reply by RaphaelFen** [reply]
Merci. 
Mais je crains qu'il n'existe pas de pendant à la fonction `tbl_uvregression()` applicable aux objets survey issus de `survey::svydesign` pour des résultats univariés (là où `tbl_regression` est remplacé par `tbl_svyregression()` pour du multivarié) ?

**Reply by larmarange** [reply]
L'aide de `tbl_uvregression()` précise que cette fonction est compatible avec des objets `survey`.

``` r
library(tidyverse)
library(labelled)
data(hdv2003, package = "questionr")
d <-
  hdv2003 |> 
  mutate(
    sexe = sexe |> fct_relevel("Femme"),
    groupe_ages = age |>
      cut(
        c(18, 25, 45, 65, 99),
        right = FALSE,
        include.lowest = TRUE,
        labels = c("18-24 ans", "25-44 ans",
                   "45-64 ans", "65 ans et plus")
      ),
    etudes = nivetud |> 
      fct_recode(
        "Primaire" = "N'a jamais fait d'etudes",
        "Primaire" = "A arrete ses etudes, avant la derniere annee d'etudes primaires",
        "Primaire" = "Derniere annee d'etudes primaires",
        "Secondaire" = "1er cycle",
        "Secondaire" = "2eme cycle",
        "Technique / Professionnel" = "Enseignement technique ou professionnel court",
        "Technique / Professionnel" = "Enseignement technique ou professionnel long",
        "Supérieur" = "Enseignement superieur y compris technique superieur"
      ) |> 
      fct_na_value_to_level("Non documenté")  
  ) |> 
  set_variable_labels(
    sport = "Pratique un sport ?",
    sexe = "Sexe",
    groupe_ages = "Groupe d'âges",
    etudes = "Niveau d'études",
    relig = "Rapport à la religion",
    heures.tv = "Heures de télévision / jour",
    poids = "Pondération de l'enquête"
  )

library(srvyr)
#> 
#> Attachement du package : 'srvyr'
#> L'objet suivant est masqué depuis 'package:stats':
#> 
#>     filter
library(survey)
#> Le chargement a nécessité le package : grid
#> Le chargement a nécessité le package : Matrix
#> 
#> Attachement du package : 'Matrix'
#> Les objets suivants sont masqués depuis 'package:tidyr':
#> 
#>     expand, pack, unpack
#> Le chargement a nécessité le package : survival
#> 
#> Attachement du package : 'survey'
#> L'objet suivant est masqué depuis 'package:graphics':
#> 
#>     dotchart
dp <- d |> 
  as_survey_design(weights = poids)

library(gtsummary)
dp |>
  tbl_uvregression(
    y = sport,
    include = c(sexe, groupe_ages, etudes, relig, heures.tv),
    method = survey::svyglm,
    method.args = list(family = quasibinomial),
    exponentiate = TRUE
  ) |> 
  bold_labels() |> 
  as_kable()
```

| **Characteristic**              |   **N**    | **OR** | **95% CI** | **p-value** |
|:--------------------------------|:----------:|:------:|:----------:|:-----------:|
| **Sexe**                        | 11,071,226 |        |            |             |
| Femme                           |            |   —    |     —      |             |
| Homme                           |            |  1.52  | 1.20, 1.91 |   \<0.001   |
| **Rapport à la religion**       | 11,071,226 |        |            |             |
| Pratiquant regulier             |            |   —    |     —      |             |
| Pratiquant occasionnel          |            |  1.23  | 0.81, 1.87 |     0.3     |
| Appartenance sans pratique      |            |  1.50  | 1.02, 2.19 |    0.039    |
| Ni croyance ni appartenance     |            |  1.83  | 1.21, 2.76 |    0.004    |
| Rejet                           |            |  1.48  | 0.78, 2.80 |     0.2     |
| NSP ou NVPR                     |            |  0.85  | 0.34, 2.16 |     0.7     |
| **Heures de télévision / jour** | 11,036,941 |  0.79  | 0.73, 0.85 |   \<0.001   |
| **Groupe d’âges**               | 11,071,226 |        |            |             |
| 18-24 ans                       |            |   —    |     —      |             |
| 25-44 ans                       |            |  0.59  | 0.39, 0.89 |    0.012    |
| 45-64 ans                       |            |  0.21  | 0.14, 0.33 |   \<0.001   |
| 65 ans et plus                  |            |  0.12  | 0.07, 0.20 |   \<0.001   |
| **Niveau d’études**             | 11,071,226 |        |            |             |
| Primaire                        |            |   —    |     —      |             |
| Secondaire                      |            |  3.96  | 2.48, 6.35 |   \<0.001   |
| Technique / Professionnel       |            |  5.41  | 3.52, 8.33 |   \<0.001   |
| Supérieur                       |            |  11.9  | 7.60, 18.5 |   \<0.001   |
| Non documenté                   |            |  28.6  | 15.1, 54.0 |   \<0.001   |

<sup>Created on 2024-03-27 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Reply by RaphaelFen** [reply]
Je n'avais pas la subtilité d'aller chercher `survey::svyglm` dans l'argument `method` de `tbl_uvregression()`.
Merci beaucoup !

**Comment by ObsTEF93** [comment]
Bonjour, 

Lorsqu'on utilise des données pondérées, les résultats affichés par broom::glance et performance::model_performance pour évaluer les performances du modèle sont-ils fiables? Ou faut-il utiliser d'autres fonctions adaptées aux données pondérées?

Lorsque j'utilise model_performance, j'ai le message suivant qui s'affiche dans la console:
1: Dans logLik.svyglm(x, ...) : svyglm not fitted by maximum likelihood.
2: Dans logLik.svyglm(x) : svyglm not fitted by maximum likelihood.
3: Dans logLik.svyglm(x, ...) : svyglm not fitted by maximum likelihood.
4: Dans logLik.svyglm(x) : svyglm not fitted by maximum likelihood.
5: Dans logLik.svyglm(x, ...) : svyglm not fitted by maximum likelihood.
6: Dans logLik.svyglm(x) : svyglm not fitted by maximum likelihood.

Merci par avance pour votre retour... et merci tout court pour la grande qualité (notamment pédagogique) de vos ressources.

**Reply by larmarange** [reply]
Normalement, si vous avez utilisez svyglm, les statistiques de glance et de performance sont correctes, car utilisent les méthodes standards fournies par survey

**Comment by elisemb1** [comment]
Bonjour, 

Je vous remercie pour ce guide qui est très précieux. Auriez-vous des exemples de code pour réaliser les "diagnostics" du modèle de régression linéaire, notamment à l'aide du package svydiags ? 

Merci pour tout !

**Reply by elisemb1** [reply]
*regression logistique :)

**Reply by larmarange** [reply]
Toutes mes excuses mais je découvre ce package. Je ne l'ai donc jamais essayé. 

Vous pouvez également regarder du package performance https://easystats.github.io/performance/index.html qui est compatible avec les modèles survey et dont la documentation est plus détaillée

**Reply by elisemb1** [reply]
Bonjour, aucun souci. J'ai finalement suivi vos conseils et utilisé le package "performance"! Un tout grand merci pour votre aide !

**Comment by titouan-ramilison-hub** [comment]
Bonjour,

Merci beaucoup pour votre travail !

Ma question concerne le chapitre "33 Régression logistique binaire pondérée", dans la partie "33.5 Prédictions marginales", vous n'indiquez pas les poids (wts) associés au survey design dans broom.helpers::plot_marginal_predictions(). Pourquoi ? Le fait d'avoir utilisé un svyglm pour estimer le modèle suffit-il pour avoir des estimations correctes sans avoir à re fournir les poids ?

**Reply by larmarange** [reply]
Bonjour et merci pour votre retour et d'avoir pointé cela.

Je viens de mettre à jour le code du chapitre : https://github.com/larmarange/guide-R/commit/0a347666da23b8cfea10f54c543120e4b10d56a0

Le version HTML sera actualisée d'ici quelques jours.

Quelques éléments de précision :

- Les prédictions du modèle restent correctes.
- Cependant, si l'on ne précise pas les poids, et dans le cas de prédictions marginales moyennes, les prédictions marginales obtenues reflètent la situation d'une population ayant une structure équivalente à la structure non pondérée de l'échantillon.
- L'ajout des poids permet d'avoir des prédictions correspondant à la structure pondérée de l'échantillon.

Bien cordialement
